invenio_drafts_resources/drafts/models.py

```diff
@@ -26,13 +27,20 @@ class DraftMetadataBase(Timestamp):
     # Enables SQLAlchemy-Continuum versioning
     __versioned__ = {}
 
-    id = db.Column(UUIDType, primary_key=True)
+    id = db.Column(
+        UUIDType,
+        primary_key=True,
+        default=uuid.uuid4,
+    )
+    """Draft identifier."""
+
+    fork_id = db.Column(UUIDType)
     """Draft identifier, it is the same than the record it is draft of"""
 
-    fork_version_id = db.Column(db.Integer, primary_key=True)
+    fork_version_id = db.Column(db.Integer)
     """Version id of the record it is draft of."""
 
-    version_id = db.Column(db.Integer, primary_key=True)
+    version_id = db.Column(db.Integer, nullable=False)
```

Comment:
Won't we need to have as a PK id + version_id?

Reply:
Nope, because we are not creating the record eagerly anymore. Therefore we cannot always use the UUID (id) of the record in the draft. As a consequence the draft has its own UUID, which is PK. Then the version_id becomes just that, a version_id. The same reason applies to the proceeding question.

Reply:
I am just thinking that, uuid even unlikely it could end up to give back an existing value:) Therefore, I think that was the case even before, we were using the tuple (uuid, version_id) so we can be sure that PK is unique when we create a draft/record :) Just for clarification, I don't mean to use any record related uuid or version ids!

Reply:
**I think there are two questions here:**

*Probability of collision of UUID*

Firstly if this is a concern, we need to change the`records` model, since it only uses a PK of UUID (no version): https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/models.py#L44

Now, statistically speaking. At first glance with 32 chars I would say the amount of produced strings is 36^32 (36 available chars power 32 chars chosen). Now google says that is 6.3340287e+49, a number I'm not even able to pronounce in terms of billions, millions, etc. I *hardly* think we will get there, and if so it means our repository is bigger than google's web index (supposed to have 130 trillion IIRC, which is around power 18).
If every person on the planet generates a new UUID4 every second, we’d expect a collision to occur after about 10 years.

*Using `version_id` for a PK*:

The problem with this statement is that in *any* case the uuid and the version id are independent. Even if the uuid collides, which I think is more than proven it won't, it can still collide with version_id. i.e two new drafts for non-existing records will get version id 1, if in addition the uuid collide... well we have a PK collision of uuid + version id, since both attributes are the same.

Reply:
Thanks for the detailed explanation :) Very interesting statistics 😅  And yes, you are right about the version id ;)